back/dora/api/serializers.py

```diff
@@ -296,6 +300,16 @@ def get_frais_autres(self, obj):
     def get_profils(self, obj):
         return [c.name for c in obj.publics.all()]
 
+    def get_publics(self, obj):
+        return list(
+            dict.fromkeys(
+                item for p in obj.publics.all() for item in p.corresponding_di_publics
+            )
+        )
```

Comment:
```suggestion
    def get_publics(self, obj):
    all_items = [
        item 
        for public in obj.publics.all()
        for item in public.corresponding_di_publics
    ]
    return list(dict.fromkeys(all_items))
```
Ta façon à faire est très maline mais je la trouve difficile à lire.

Reply:
Modif faite ceb526a80916ab0a0c4a531fc5b11a31a47bd430